client/client.go

```diff
@@ -9,19 +9,39 @@ import (
 	"github.com/docker/docker-credential-helpers/credentials"
 )
 
+// isValidCredsMessage checks if 'msg' contains invalid credentials error message.
+// It returns whether the logs are free of invalid credentials errors and the error if it isn't.
+// error values can be errCredentialsMissingServerURL or errCredentialsMissingUsername.
+func isValidCredsMessage(msg string) (bool, error) {
+	if credentials.IsCredentialsMissingServerURLMessage(msg) {
+		return false, credentials.NewErrCredentialsMissingServerURL()
+	}
+
+	if credentials.IsCredentialsMissingUsernameMessage(msg) {
+		return false, credentials.NewErrCredentialsMissingUsername()
+	}
+
+	return true, nil
+}
+
 // Store uses an external program to save credentials.
-func Store(program ProgramFunc, credentials *credentials.Credentials) error {
+func Store(program ProgramFunc, creds *credentials.Credentials) error {
 	cmd := program("store")
 
 	buffer := new(bytes.Buffer)
-	if err := json.NewEncoder(buffer).Encode(credentials); err != nil {
+	if err := json.NewEncoder(buffer).Encode(creds); err != nil {
 		return err
 	}
 	cmd.Input(buffer)
 
 	out, err := cmd.Output()
 	if err != nil {
 		t := strings.TrimSpace(string(out))
```

Comment:
for a followup PR: it might be useful to combine the  `cmd.Output()`, `strings.TrimSpace`, and `isValidCredsMessage` parsing into a shared function that all of these exported functions can use.
The same functionality seems to be replicated across `Get`, `Store`, `Erase`, etc.